compiler/rustc_parse/src/parser/diagnostics.rs

```diff
@@ -400,6 +401,35 @@ impl<'a> Parser<'a> {
             .map(|x| TokenType::Token(x.clone()))
             .chain(inedible.iter().map(|x| TokenType::Token(x.clone())))
             .chain(self.expected_tokens.iter().cloned())
+            .filter_map(|token| {
+                // filter out suggestions which suggest the same token which was found and deemed incorrect
+                fn is_ident_eq_keyword(found: TokenKind, expected: TokenType) -> bool {
```

Comment:
This can get arguments by ref, to not to clone things.

Reply:
Done.